### DISCUSSION TITLE
RPGLE snippets - help wanted!

### DISCUSSION THREAD

**Discussion by worksofliam** [codefori/vscode-ibmi#926] [Ideas]
Hi

We've been making use of RPGLE snippets for a long time. Originally, we only shipped snippets for build-in functions, but this morning I went ahead and added some for other RPG things (like `dcl-s`).

What are other snippets you'd like to see to make developing faster?

Feel free to leave a reply... or a PR!

https://github.com/halcyon-tech/vscode-rpgle/blob/main/schemas/rpgle.code-snippets#L15

**Comment by andreas-prouza** [comment]
Dear @worksofliam,
Thanks for your extensions!

- dcl-f (display files or PF/LF)
- snd-msg (IBM i 7.5)
- for-each (IBM i 7.5)
- dsply (with message queue and answer variable
- exec sql Whenever SQLERROR do {rpg_proc()};
- exec sql Whenever SQLWARNING do {rpg_proc()};
- main-proc
I usually add a main-proc to a rpgle programm:
```
dcl-proc main;
  dcl-pi *n;
  end-pi;

  return;
end-prod;
```
Many thanks!
Andi

**Comment by beckhamrryyaann** [comment]
add all closing operators, like
* endif
* enddo
* endfor
* endsl
* etc
